src/System.Device.Gpio/System/Device/Gpio/ThrowHelper.cs

```diff
@@ -0,0 +1,91 @@
+        }
+
+        public static PlatformNotSupportedException GetPlatformNotSupportedException(ExceptionResource resource)
+        {
+            return new PlatformNotSupportedException(GetResourceString(resource, -1, -1));
+        }
+
+        private static IOException GetIOException(string message)
+        {
+            return new IOException(message);
+        }
+
+        private static InvalidOperationException GetInvalidOperationException(string message)
+        {
+            return new InvalidOperationException(message);
+        }
+
+        public static InvalidOperationException GetInvalidOperationException(ExceptionResource resource, int pin = -1)
+        {
+            return GetInvalidOperationException(GetResourceString(resource, -1, pin));
+        }
+
+        private static string GetResourceString(ExceptionResource resource, int errorCode, int pin)
+        {
+            string message = "";
+            switch (resource)
+            {
+                case ExceptionResource.NoChipIteratorFound:
+                    message = $"Unable to find a chip iterator, error code: { errorCode}";
+                    break;
+                case ExceptionResource.NoChipFound:
+                    message = $"Unable to find a chip, error code: {errorCode}";
+                    break;
+                case ExceptionResource.PinModeReadError:
+                    message = $"Error while reading pin mode for pin: {pin}";
+                    break;
+                case ExceptionResource.OpenPinError:
+                    message = $"Pin {pin} not available, error: {errorCode}";
+                    break;
+                case ExceptionResource.ReadPinError:
+                    message = $"Error while reading value from pin: {pin}, error: {errorCode}";
+                    break;
+                case ExceptionResource.PinNotOpenedError:
+                    message = $"Can not access pin {pin} that is not open.";
+                    break;
+                case ExceptionResource.SetPinModeError:
+                    message = $"Error setting pin mode, for pin: {pin}, error: {errorCode}";
+                    break;
+                case ExceptionResource.ConvertPinNumberingSchemaError:
+                    message = $"This driver is generic so it cannot perform conversions between pin numbering schemes.";
+                    break;
+                default:
+                    Debug.Fail($"The ExceptionResource enum value: {resource} is not part of the switch. Add the appropriate case and exception message.");
+                    break;
+            }
+            
+            return message;
+        }
+
+        internal enum ExceptionResource
+        {
+            NoChipIteratorFound,
+            NoChipFound,
+            PinModeReadError,
+            OpenPinError,
+            ReadPinError,
+            PinNotOpenedError,
+            SetPinModeError,
+            ConvertPinNumberingSchemaError
```

Comment:
nit: It would be cleaner if this enum was outside the ThrowHelper static class. That way you can remove the boilerplate `ThrowHelper.ExceptionResource....` everywhere and use it like a regular enum: `ExceptionResource....`